src/build_info.cpp

```diff
@@ -550,6 +550,10 @@ list_formatter video_settings_report_internal(const std::string& heading = "")
 		placeholder = "Running in non-interactive mode.";
 	}
 
+	if(!video::has_window()) {
```

Comment:
Should this be an `else if` of the `if(video::headless())`?

Reply:
It doesn't really matter.

Reply:
I suppose you could argue that the latter message will clobber the previous one if Wesnoth is really running in headless mode. On the other hand the odds that `--report` will be used and relied upon while explicitly in headless mode seem quite slim, which kinda makes the headless case redundant.

Reply:
No. It's not important.

My opinion on this has changed in the previous 30 minutes, after investigating why `--report` doesn't imply headless mode anyway. Using `--report` gets processed in `process_command_args`, and then there's a mandatory quit afterwards. So I think there's no codepath that actually gets to here with `video::headless()==true`.